python/lsst/ap/association/trailedSourceFilter.py

```diff
@@ -0,0 +1,106 @@
+# This file is part of ap_association.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ("TrailedSourceFilterTask", "TrailedSourceFilterConfig")
+
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+from lsst.utils.timer import timeMethod
+
+
+class TrailedSourceFilterConfig(pexConfig.Config):
+    """Config class for TrailedSourceFilterTask.
+    """
+    maxTrailLength = pexConfig.Field(
+        dtype=float,
+        doc='Maximum trail length permitted is less than 10 degrees/day. '
+            'This is a rate of 0.416 arcseconds per second. As trail length'
+            ' is measured in arcseconds, it is dependant on the length of '
+            'the exposure.',
+        default=0.416,
+    )
+
+
+class TrailedSourceFilterTask(pipeBase.Task):
+    """Find trailed sources in DIASources. This task checks the length of
+    trailLength in the DIASource catalog using a given
+     arcsecond/second rate from maxTrailLength and the exposure time. The
+     two values are used to calculate the maximum allowed trail length and
+     filters out any trail longer than the maximum.
+
+    """
+
+    ConfigClass = TrailedSourceFilterConfig
+    _DefaultName = "trailedSourceFilter"
+
+    @timeMethod
+    def run(self, dia_sources, diffIm):
+        """Find trailed sources which do not meet requirements and
+        will not be included in the diaSource catalog.
+
+        Parameters
+        ----------
+        dia_sources : `pandas.DataFrame`
+            New DIASources to be checked for trailed sources.
+        diffIm : `lsst.afw.image.ExposureF`
+            Difference image on which the DIASources were detected.
+
+        Returns
+        -------
+        result : `lsst.pipe.base.Struct`
+            Results struct with components.
+
+            - ``dia_sources`` : DIASource table that is free from unwanted
+             trailed sources. (`pandas.DataFrame`)
+
+            - ``trailed_dia_sources`` : DIASources that have trailed more
+            than 0.416 arcseconds/second*exposure_time. (`pandas.DataFrame`)
```

Comment:
```suggestion
            than maxTrailLength*exposure_time. (`pandas.DataFrame`)
```

We don't want to include numbers here, since it's configurable.

Reply:
Updated with maxTrailLength